subprojects/execution/src/test/groovy/org/gradle/internal/execution/impl/steps/CacheStepTest.groovy

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.impl.steps
+
+import com.google.common.collect.ImmutableSortedMap
+import org.gradle.caching.internal.command.BuildCacheCommandFactory
+import org.gradle.caching.internal.controller.BuildCacheController
+import org.gradle.caching.internal.origin.OriginMetadata
+import org.gradle.internal.execution.CacheHandler
+import org.gradle.internal.execution.ExecutionOutcome
+import org.gradle.internal.execution.OutputChangeListener
+import org.gradle.internal.execution.UnitOfWork
+import org.gradle.internal.fingerprint.CurrentFileCollectionFingerprint
+import org.gradle.internal.fingerprint.impl.EmptyCurrentFileCollectionFingerprint
+import org.gradle.internal.id.UniqueId
+import spock.lang.Specification
+
+import javax.annotation.Nullable
+
+class CacheStepTest extends Specification {
+
+    def buildCacheController = Mock(BuildCacheController)
+    def buildCacheCommandFactory = Mock(BuildCacheCommandFactory)
+    def outputChangeListener = Mock(OutputChangeListener)
+    def currentBuildId = UniqueId.generate()
+    Step<Context, CurrentSnapshotResult> delegateStep = Mock(Step)
+    def cacheStep = new CacheStep<CachingContext>(buildCacheController, outputChangeListener, buildCacheCommandFactory, delegateStep)
+    def cacheHandler = Mock(CacheHandler)
+    def unitOfWork = Mock(UnitOfWork)
+    def loadMetadata = Mock(BuildCacheCommandFactory.LoadMetadata)
+    def cachedOriginMetadata = Mock(OriginMetadata)
+    def context = Stub(CachingContext) {
+        getCacheHandler() >> cacheHandler
+        getWork() >> unitOfWork
+    }
```

Comment:
This could be a proper type, too, without much more code. That would make it more refactor-safe, too.

Reply:
Done.